### PR TITLE
Feature/ci cd

### PR DESCRIPTION
**Pull request by sagar-wednesday**
### Ticket Link

---

### Related Links

---

### Description

---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### Live Link

---

- http://wednesday-react-template.s3-website.ap-south-1.amazonaws.com/<BRANCH_NAME>

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Get branch name
+        id: vars
+        run: echo ::set-output name=branch::${{ github.event.pull_request.head.ref }}
```
Comment (praveenkumar1798) [inline .github/workflows/cd.yml]: This is for every branch you create. You need to change it to work only for master

**Review by praveenkumar1798 — APPROVED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (sagar-wednesday) [reply .github/workflows/cd.yml]: okay

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
-name: React Template CD
+name: ITUNES React CD
 on:
-  push:
+  pull_request:
```
Comment (praveenkumar1798) [inline .github/workflows/cd.yml]: You dont need this

Comment (sagar-wednesday) [reply .github/workflows/cd.yml]: yeah when I'm done with testing I will change it to push for the master

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -14,33 +18,35 @@ jobs:
...
-          branch: ${{ steps.vars.outputs.stage }}
-          unprotect_reviews: true
+          args: --acl public-read --follow-symlinks
+        env:
+          SOURCE_DIR: './build/'
+          AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+          PATHS: '/*'
```
Comment (praveenkumar1798) [inline .github/workflows/cd.yml]: set the env once at the top instead of setting it up seperately for deploy and invalidation

Comment (sagar-wednesday) [reply .github/workflows/cd.yml]: okay I'll try that

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -14,33 +24,34 @@ jobs:
...
+      - name: AWS S3 Deploy
+        uses: jakejarvis/s3-sync-action@v0.5.0
         with:
-          token: ${{ secrets.PUSH_TO_PROTECTED_BRANCH }}
-          branch: ${{ steps.vars.outputs.stage }}
-          unprotect_reviews: true
+          args: --acl public-read --follow-symlinks
+        env:
+          SOURCE_DIR: './build/'
+          DEST_DIR: ${{ steps.vars.outputs.branch }}
```
Comment (praveenkumar1798) [inline .github/workflows/cd.yml]: no need to have it here. Put it at top

**Review by praveenkumar1798 — APPROVED** [review]

**Pull request merged**